internal/worker/peergrouper/desired.go

```diff
@@ -595,55 +568,20 @@ func (p *peerGroupChanges) updateAddressesFromInternal() error {
 		// address has changed, we enforce the policy of requiring a
 		// configured HA space when there are multiple cloud-local addresses.
 		if !unchanged {
-			multipleAddresses = append(multipleAddresses, id)
-			if err := m.host.SetStatus(getStatusInfo(multiAddressMessage)); err != nil {
-				return errors.Trace(err)
-			}
+			member.Address = selectSingleAddress(addrs)
+			p.desired.isChanged = true
 		}
 	}
 
-	if len(multipleAddresses) > 0 {
-		ids := strings.Join(multipleAddresses, ", ")
-		return fmt.Errorf("juju-ha-space is not set and these nodes have more than one usable address: %s"+
-			"\nrun \"juju controller-config juju-ha-space=<name>\" to set a space for Mongo peer communication", ids)
-	}
 	return nil
 }
 
-// updateAddressesFromSpace updates the member addresses based on the
-// configured HA space.
-// If no addresses are available for any of the nodes, then such nodes
-// have their status set and are included in the detail of the returned error.
-func (p *peerGroupChanges) updateAddressesFromSpace() error {
-	space := p.info.haSpace
-	var noAddresses []string
-
-	for _, id := range p.sortedMemberIds() {
-		m := p.info.controllers[id]
-		addr, err := m.SelectMongoAddressFromSpace(p.info.mongoPort, space)
-		if err != nil {
-			if errors.Is(err, errors.NotFound) {
-				noAddresses = append(noAddresses, id)
-				msg := fmt.Sprintf("no addresses in configured juju-ha-space %q", space.Name)
-				if err := m.host.SetStatus(getStatusInfo(msg)); err != nil {
-					return errors.Trace(err)
-				}
-				continue
-			}
-			return errors.Trace(err)
-		}
-		if addr != p.desired.members[id].Address {
-			p.desired.members[id].Address = addr
-			p.desired.isChanged = true
-		}
-	}
-
-	if len(noAddresses) > 0 {
-		ids := strings.Join(noAddresses, ", ")
-		return fmt.Errorf(
-			"no usable Mongo addresses found in configured juju-ha-space %q for nodes: %s", space.Name, ids)
-	}
-	return nil
+// selectSingleAddress selects only one address from list of addresses. The
+// selection is done in a consistent fashion.
+func selectSingleAddress(addrs []string) string {
+	// FIXME: this should first sort addresses (or filter them) according
+	// to some criteria (which?).
+	return addrs[0]
```

Comment:
@manadart @SimonRichardson is selecting the first address, without any sorting (so therefore not consistently) good enough? I see that the addresses that we get here are already filtered to match cloud local scope so maybe it's good enough